sklearn/cluster/tests/test_affinity_propagation.py

```diff
@@ -185,7 +185,7 @@ def test_affinity_propagation_predict_non_convergence():
 def test_affinity_propagation_non_convergence_regressiontest():
     X = np.array([[1, 0, 0, 0, 0, 0], [0, 1, 1, 1, 0, 0], [0, 0, 1, 0, 0, 1]])
     af = AffinityPropagation(affinity="euclidean", max_iter=2, random_state=34).fit(X)
-    assert_array_equal(np.array([-1, -1, -1]), af.labels_)
+    assert_array_equal(np.array([0, 0, 0]), af.labels_)
```

Comment:
This is a backward incompatible change.

If we want to remain backward compatible, we would need to add a new parameter to `AffinityPropagation` that triggers the new behavior. Something like: `assign_clusters_if_not_converged`.

WDYT? @GaelVaroquaux @jeremiedbb